src/Plugin/CaptureUtility/WebPageTestCaptureUtility.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Drupal\performance_budget\Plugin\CaptureUtility;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\web_page_archive\Plugin\ConfigurableCaptureUtilityBase;
+use Drupal\web_page_archive\Plugin\CaptureResponse\UriCaptureResponse;
+use WidgetsBurritos\WebPageTest\WebPageTest;
+
+/**
+ * Skeleton capture utility, useful for creating new plugins.
+ *
+ * @CaptureUtility(
+ *   id = "pb_wpt_capture",
+ *   label = @Translation("Web page test capture utility", context = "Web Page Archive"),
+ *   description = @Translation("Runs url through webpagetest.org.", context = "Web Page Archive")
+ * )
+ */
+class WebPageTestCaptureUtility extends ConfigurableCaptureUtilityBase {
+
+  /**
+   * Most recent response.
+   *
+   * @var string|null
+   */
+  private $response = NULL;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function capture(array $data = []) {
+    // Configuration data is stored in $this->configuration. For example:
+    $wpt_api = $this->configuration['wpt_api'];
+    $wpt = new WebPageTest($wpt_api);
+    $state_key = $this->getStateKey($data);
+    $test_id = $this->state()->get($state_key);
+    $response_content = '';
+    if (!isset($test_id)) {
+      if ($response = $wpt->runTest($data['url'])) {
+        if ($response->statusCode == 200) {
+          $this->state()->set($state_key, $response->data->testId);
+        }
+      }
+
+      $this->response = NULL;
+    }
+    else {
+      if ($response = $wpt->getTestStatus($test_id)) {
+        if ($response->statusCode == 200) {
+          // Test is complete.
+          if ($response = $wpt->getTestResults($test_id)) {
+            $file_path = \Drupal::service('file_system')->realpath(file_default_scheme() . "://");
+            $save_dir = "{$file_path}/performance-budget/wpt/{$data['web_page_archive']->id()}/{$data['run_uuid']}";
+            $file_name = preg_replace('/[^a-z0-9]+/', '-', strtolower($data['url'])) . '.json';
+            $file_location = "{$save_dir}/{$file_name}";
+
+            if (!file_prepare_directory($save_dir, FILE_CREATE_DIRECTORY | FILE_MODIFY_PERMISSIONS)) {
+              throw new \Exception("Could not write to $save_dir");
+            }
+            file_put_contents($file_location, json_encode($response->data));
```

Comment:
@pobtastic Right now this is a copy/paste from the HTML capture utility. There is a separate task to address file stream wrappers, throughout WPA: https://www.drupal.org/node/2901781 

Grant it, it's a separate module, but I'd prefer deferring that task until then, so we can consistently fix all of them. Are you okay with that?